### Python Data Type

Do you remember in Math class where there are different type of numbers? There are <font color=green>**Integer**</font> or <font color=green>**Whole Numbers**</font>, <font color=green>**Real Number**</font>. In Python 3 and other programming language there is are many other data type, for now we will keep it simple and list some of the more common ones. 

#### Here is a list of some common Data Type 
<div class="alert alert-block alert-info">
-  <font color=blue>**Character**</font>
    -  Think of character as any <font color=green>**single key**</font> on your keyboard. This could be numbers, letters and special characters.

-  <font color=blue>**String**</font>
    -  String is any combination of Characters encased between <font color=green>**" "**</font> or <font color=green>**' '**</font>. This can include <font color=green>**numbers**</font>, <font color=green>**letters**</font> and even <font color=green>**special characters**</font>.
        -  Remember  <font color=green>**"/""**</font> and  <font color=green>**"/'"**</font> can be used interchangably however if you choose to use one over the other, you have to be <font color=blue>**consistent**</font>! So you <font color=red>cannot have **"Hello'** </font>but  <font color=green> **"Hello"** or **'Hello'** </font> are exactly the same
-  <font color=blue>**Integer**</font>
    -  Refers to whole numbers like  <font color=green>**3**</font> or  <font color=green>**262361508**</font>.
-  <font color=blue>**Float**</font> 
    -  Refers to decimal numbers like  <font color=green>**3.14**</font>.
    
</div>
We can see the difference in the example below:

In [1]:
This_Refers_To_The_Integer_4 = 4
This_Refers_To_The_String_4 = '4'

As you can see above the two <font color=blue>**4**</font> are displayed in different colours. One is <font color=green>green for the **Integer**</font> and the other is <font color=red>red for the **String**</font>. This is to indicate that <font color=blue>**they are different**</font>. Depending on what program you are using to write Python code it may be a different colour or have no colour atgreen.

Now if we do some basic operation on the two variables or nicknames we can see they give us different results

In [2]:
This_Refers_To_The_Integer_4 = 4

This_Refers_To_The_Integer_4 * 4

16

In [3]:
This_Refers_To_The_String_4 = '4'

This_Refers_To_The_String_4 * 4

'4444'

As you can see they give us different result because they are different data type. One multipled the number 4 x 4 resulting in 16. In the second case we are saying we want "4 copies of the character 4" resulting in "4444"

Now that we have discussed the different data types, we are going to touch on Data Structures